include/eosio/vm/host_function.hpp

```diff
@@ -17,555 +20,353 @@
-
-      // Error type to hopefully make a somewhat understandable error message when
-      // the user did not provide a correct overload of from_wasm.
-      template<typename T, typename... Tail>
-      struct no_viable_overload_of_from_wasm {};
-
-      // Searches for a match of from_wasm following the principal of maximum munch.
-      template<std::size_t N, typename T, typename Tail>
-      constexpr auto make_value_getter_impl() {
-         if constexpr(std::is_same_v<try_value_getter_t<N, T, Tail>, void>) {
-            if constexpr (N == 0) {
-               return no_viable_overload_of_from_wasm<T>{};
-            } else {
-               return make_value_getter_impl<N-1, T, Tail>();
-            }
-         } else {
-            return try_value_getter_t<N, T, Tail>{};
+      template <typename Args, class Type_Converter>
+      static inline constexpr std::size_t total_operands_v = total_operands<Args, 0, Type_Converter>();
+
+      template <typename Args, typename S, std::size_t At, class Type_Converter, std::size_t... Is>
+      inline constexpr decltype(auto) create_value(Type_Converter& tc, std::index_sequence<Is...>) {
+         constexpr std::size_t offset = total_operands_v<Args, Type_Converter> - 1;
+         if constexpr (!std::is_same_v<no_match_t, std::decay_t<decltype(tc.template as_value<S>(pop_value<offset - At>(tc)))>>)
+            return tc.template from_wasm<S>( tc.template as_value<S>(pop_value<offset - At>(tc)) );
+         else {
+            return tc.template from_wasm<S>(pop_value<offset - (At + Is)>(tc)...);
          }
       }
 
-      template<typename T, typename Tail>
-      constexpr auto make_value_getter() {
-         return make_value_getter_impl<Tail::size, T, Tail>();
+      template <typename S, typename Type_Converter>
+      inline constexpr std::size_t skip_amount(Type_Converter& tc) {
+         if constexpr (!std::is_same_v<no_match_t, std::decay_t<decltype(tc.template as_value<S>(pop_value<0>(tc)))>>)
+            return 1;
+         else
+            return std::tuple_size_v<from_wasm_type_deducer_t<Type_Converter, S>>;
       }
 
-      // args should be a tuple
-      // Constructs a cons of the translated arguments given a list of
-      // argument types, a wasm stack, and a wasm allocator.
-      template<typename Args>
-      struct pack_args;
+      template <typename Args, std::size_t At, class Type_Converter>
+      inline constexpr decltype(auto) get_values(Type_Converter& tc) {
+         if constexpr (At >= std::tuple_size_v<Args>)
+            return std::tuple<>{};
+         else {
+            using source_t      = std::tuple_element_t<At, Args>;
+            constexpr size_t skip_amt = skip_amount<source_t>(tc);
+            return std::tuple_cat(std::tuple<source_t>{create_value<Args, source_t, At>(tc, std::make_index_sequence<skip_amt>{})}, get_values<Args, At + skip_amt>(tc));
```

Comment:
I'm confused about what `At` represents here.  The result of get_values will ultimately be passed to std::invoke, right?  That means that the size of the result must match the number of arguments of the intrinsic.  Args is the intrinsic's arguments, so skipping elements can't possibly work.  What am I missing?

Reply:
`At` represents the index in the argument list that you are at.  We need a way to skip arguments that have already been pulled from the operand stack.

Reply:
My comment is obsolete.  You already fixed the problem by separating At and Skip.